src/services/formatting/formatting.ts

```diff
@@ -3,6 +3,7 @@ namespace ts.formatting {
     export interface FormatContext {
         readonly options: FormatCodeSettings;
         readonly getRules: RulesMap;
+        readonly host?: FormattingHost;

```

Comment:
Why is this optional ? Should t it always be there to ensure that any new line related operations are handled correctly

Reply:
I made it optional because I was thinking it was a public API, and a programmatic user really doesn’t need to supply this because they can just set `options.newLineCharacter`. But since it’s internal, I think it’s fine to put it back to required.